### PR TITLE
Pass command line arguments to "yarn start"

### PR DESCRIPTION
**Pull request by AniruddhaTayade123**
1. Using yargs library
2. Valid flags are "live" and "file"
3. If both flags are not provided then exit the server.

### CODE REVIEW

**Comment by AniruddhaTayade123** [comment]
@gabhijit  @mayurhyphenOs 
Please review the changes.

File: package.json [hunk 1]
```diff
@@ -9,6 +9,6 @@
   "dependencies": {
     "JSONStream": "^1.3.5",
     "ws": "^7.3.0",
-    "redis": "^3.0"
```
Comment (gabhijit) [inline package.json:12]: Why is this dependency removed? It is required for the operation of the program subsequently.

File: server.js [hunk 2]
```diff
@@ -68,7 +76,7 @@ const socketServer = new WebSocket.Server({ port: 3030 });
 console.log("listening on 3030");
 
 // FIXME: Comment the following line when you want to test with PCAP file
```
Comment (gabhijit) [inline server.js:83]: If `--live` flag is passed above, the following line needs to be uncommented.

File: server.js [hunk 3]
```diff
@@ -68,7 +76,7 @@ const socketServer = new WebSocket.Server({ port: 3030 });
...
-// pktStreamer.emit("packet");
+// pktStreamer.emit("packet"); 
 
 socketServer.on("connection", (socketClient) => {
```
Comment (gabhijit) [inline server.js:88]: Also below line number - This should be wrapped inside an if loop if `--live` is false.

**Review by gabhijit — CHANGES_REQUESTED** [review]
Aniruddha, 

This is a good start. It will help to get your pull request merged if you follow following guidelines - 

1. Try running the program even without your changes. You should still see the `parseJSON` function called.

2. Your couple of changes are correct - but looking at these changes does not give sufficient confidence that the code is likely to work.

3. Note: the expected behavior is as follows - 
   - if `--live <interface-name>` is mentioned - it should subscribe to a Redis queue and start picking packets from the Redis queue.
   - if `--file <filename>` is given - your `parseJSON` function should parse appropriate file (that was passed on the command line and also error out if the passed file cannot be opened and/or parsed.

I guess the description of the issue is perhaps not complete enough for you to have understood those things. This might have caused the problem. 

Let's try to get it working and come to a point where we'll be able to merge in the repository. I'll provide as much help as possible for you to get there. 

Hope this helps.

**Comment by AniruddhaTayade123** [comment]
@gabhijit 
I will start working on the above mentioned functionality.

**Review by gabhijit — CHANGES_REQUESTED** [review]
There is a bit of a problem with how you are parsing the args - 

If I specify - `yarn server --file` (without the required argument) or `yarn start --live` without the required argument, The correct interface name is not selected as these arguments are treated as `flags` and evaluate to `true` and the code continues. 

This is not a correct behavior. Look at the following - http://yargs.js.org/docs/ - to see how to make sure both `--live` and `--file` take arguments (in particular look at the [`coerce`](http://yargs.js.org/docs/#api-reference-coercekey-fn) for the file argument. For the interface argument - unfortunately there is no easy way to check right now, so let it be. Just make sure that at-least 'some' interface name is passed.

File: server.js [hunk 4]
```diff
@@ -4,6 +4,19 @@ const JSONStream = require("JSONStream");
...
 const { promisify } = require("util");
+const args = require('yargs').argv;
+
+const { live, file } = args;
+if( live === undefined && file === undefined){
+  console.log("No flag provided. Please provide a valid flag as instucted below.")
+  console.log("Usage: yarn start --live <interface-name> | --file <path-to-pcap-filename>");
+  process.exit(1)
+}
+if( live && file ){
```
Comment (gabhijit) [inline server.js:15]: This is correct - but this is treating both `--live` and `--file` as flags, but they should take extra parameter (mandatory) and used that.

**Review by gabhijit — CHANGES_REQUESTED** [review]

File: server.js (hunk 4 above)
Comment (AniruddhaTayade123) [reply server.js:15]: Okay, I see.
Will go through the documentation and try to make required changes.

**Comment by gabhijit** [comment]
@AniruddhaTayade123 : It's been over 10 days on this and right now I want to make additional changes. So I am going ahead and making the fix myself.

**Comment by AniruddhaTayade123** [comment]
@gabhijit 
Sir, I went through the documentation and understood that part theoretically though I could not figure out how to resolve the issue by implementing it in the code. I am still thinking how I can make it work out.

**Comment by gabhijit** [comment]
Okay. Likely I will work on it tomorrow and fix this. Because I am a little blocked by this at the moment. Just wanted to check with you if there is any progress.

**Comment by gabhijit** [comment]
The required code is now in master. Closing this PR.

**Pull request closed without merging**